models/source_list.py

```diff
@@ -506,6 +506,34 @@ def calc_aper_cor( self, aper_num=0, inf_aper_num=None, min_stars=20 ):
 
         return -2.5 * np.log10( meanrat )
 
+    def estimate_lim_mag(self, aperture=1, givePlotParams=False):
+        # if aperture = -1: using psf mags, aperture defaults to 1
+        # image must also have zero point
+        if aperture >= 0 and self.zp != None:
+            aperCorr = self.calc_aper_cor(aperture)
+            zeroPoint = self.zp.zp
+            flux, fluxerr = self.apfluxadu(aperture)
+            mags = -2.5 * np.log10(flux) + zeroPoint + aperCorr
+            snr = flux/fluxerr
+            mask = (snr >= 3) & (snr <= 20) #only fitting for sources 3 < SNR < 20 
+            snrMasked = np.log(snr[mask]) #log snrs
+            magsMasked = mags[mask]
+
+            m,c = np.polyfit(snrMasked,magsMasked,1) #calculate slope and intercept of fitted line
+            limMagEst = m * np.log(5) + c #limiting magnitude estimate at SNR = 5
+
+            if givePlotParams:
+                return limMagEst, snrMasked, magsMasked, m, c
+            else:
+                return limMagEst
+                
+        else:
+            #Will not provide limiting magnitude estimate if using PSF photometry 
+            #or if no zero point
```

Comment:
Remind me why we can't have limiting magnitude based on PSF photometry?

Reply:
I think this is just what we decided to do, based on the fact that PSF photometry in the pipeline is currently unreliable.

Reply:
I think we should still allow limiting magnitude based on it, if someone wants it, just not have it as a default.